mqr/tests/test_baseline_ussd.py

```diff
@@ -1,9 +1,150 @@
+            {
+                "results": contacts,
+                "next": None,
+            },
+            status=200,
+        )
+
+    client = await sanic_client(app)
+    url = config.RAPIDPRO_URL
+    config.RAPIDPRO_URL = f"http://{client.host}:{client.port}"
+    config.RAPIDPRO_TOKEN = "testtoken"
+    yield client
+    config.RAPIDPRO_URL = url
+
+
+@pytest.mark.asyncio
+async def test_state_survey_start(rapidpro_mock):
+    u = User(addr="27820001003", state=StateData())
+    app = Application(u)
+    msg = Message(
+        content=None,
+        to_addr="27820001002",
+        from_addr="27820001003",
+        transport_name="whatsapp",
+        transport_type=Message.TRANSPORT_TYPE.HTTP_API,
+        session_event=Message.SESSION_EVENT.NEW,
+    )
+    [reply] = await app.process_message(msg)
+    assert len(reply.content) < 140
+    assert reply.content == "\n".join(
+        [
+            "1/13",
+            "",
+            "Do you plan to breastfeed your baby after birth?",
+            "1. Yes",
+            "2. No",
+            "3. Skip",
+        ]
+    )
+
+    assert [r.path for r in rapidpro_mock.app.requests] == ["/api/v2/contacts.json"]
+
+
+@pytest.mark.asyncio
+async def test_state_survey_start_not_mqr_contact(rapidpro_mock):
+    u = User(addr="27820001004", state=StateData())
+    app = Application(u)
+    msg = Message(
+        content=None,
+        to_addr="27820001002",
+        from_addr="27820001004",
+        transport_name="whatsapp",
+        transport_type=Message.TRANSPORT_TYPE.HTTP_API,
+        session_event=Message.SESSION_EVENT.NEW,
+    )
+    [reply] = await app.process_message(msg)
+    assert len(reply.content) < 140
+    assert reply.content == "TODO COPY"
+
+    assert [r.path for r in rapidpro_mock.app.requests] == ["/api/v2/contacts.json"]
+
+
+@pytest.mark.asyncio
+async def test_state_survey_start_not_found(rapidpro_mock):
+    u = User(addr="27820001005", state=StateData())
+    app = Application(u)
+    msg = Message(
+        content=None,
+        to_addr="27820001002",
+        from_addr="27820001005",
+        transport_name="whatsapp",
+        transport_type=Message.TRANSPORT_TYPE.HTTP_API,
+        session_event=Message.SESSION_EVENT.NEW,
+    )
+    [reply] = await app.process_message(msg)
+    assert len(reply.content) < 140
+    assert reply.content == "TODO COPY"
+
+    assert [r.path for r in rapidpro_mock.app.requests] == ["/api/v2/contacts.json"]
+
+
+@pytest.mark.asyncio
+async def test_state_start_temporary_errors(rapidpro_mock):
+    rapidpro_mock.app.errormax = 3
+    u = User(addr="27820001001", state=StateData())
+    app = Application(u)
+    msg = Message(
+        content=None,
+        to_addr="27820001002",
+        from_addr="27820001001",
+        transport_name="whatsapp",
+        transport_type=Message.TRANSPORT_TYPE.HTTP_API,
+        session_event=Message.SESSION_EVENT.NEW,
+    )
+    [reply] = await app.process_message(msg)
```

Comment:
Should we assert something on this reply? Maybe the content matches the error message we expect.